### PR TITLE
Linter cleanup

### PR DESCRIPTION
**Pull request by gregsvo**
# Things Done
* Changed port to be able to be dynamically set (heroku needed)
* Changed string concatenation to more modern usage. ie: 
```javascript
RegExp('(^| )' + classNameRegex + '( |$)', 'gi').test(target.className);

// is now:

RegExp(`(^| )${classNameRegex}( |$)`, 'gi').test(target.className);
```
* Standardized Spacing
* Added linter warnings:
  * eol-last : *Blank line at end of file*
  *  indent: *['error', 4, {'SwitchCase': 1}] : *indent warnings*
  * 'no-trailing-spaces': 1 : *No Trailing Spaces*
  * no-extra-parens': 1 : *No Extra Parens*
  *  eqeqeq: 1: *Require === and !== (eqeqeq)*

# How To Test
If builds, it's working.

### CODE REVIEW

File: src/js/mount.js [hunk 1]
```diff
@@ -1,4 +1,4 @@
 import SBWidget from './widget';
 
 let sb = new SBWidget();
-sb.start('9DA1B1F4-0BE6-4DA8-82C5-2E81DAB56F23');
+sb.start('0CE2A788-CE39-4800-AF78-65738A46D687');
```
Comment (samsawan) [inline src/js/mount.js]: I think it's a good idea to keep this app id to the default one SB provides for us; we're publishing this up on npm and probably wanna keep this secret

File: Procfile [hunk 2]
```diff
@@ -0,0 +1 @@
+web: npm start
```
Comment (samsawan) [inline Procfile]: I think you should add the `Procfile` to the `.npmignore` so we don't publish this along with the package.

File: package.json [hunk 3]
```diff
@@ -2,12 +2,16 @@
...
+  "engines": {
+    "node": "9.5.0"
+  },
   "main": "index.js",
   "scripts": {
     "prod-build": "webpack -p",
     "start-dev": "webpack-dev-server",
     "start": "webpack -p && node test_server/server.js",
     "test": "echo \"Error: no test specified\" && exit 1",
+    "postinstall": "webpack --config ./webpack.config.js --progress --colors",
```
Comment (samsawan) [inline package.json]: What does this do?

File: test_server/server.js [hunk 4]
```diff
@@ -1,7 +1,7 @@
 const express = require('express');
 const app = express();
 
-const PORT = 9001;
+const PORT = process.env.PORT || 9001
```
Comment (samsawan) [inline test_server/server.js]: Semicolon at the end of the line here.

Is this logic used in the Heroku app?

File: Procfile [hunk 5]
```diff
@@ -1 +0,0 @@
-web: npm start
```
Comment (samsawan) [inline Procfile]: if this is gone we can remove it from the `.npmignore` file

Comment (gregsvo) [reply Procfile]: god dammit.

**Pull request merged**